### DISCUSSION TITLE
How do I change the alignment of an input label?

### DISCUSSION THREAD

**Discussion by storyteller-dev** [ctreffe/alfred#138] [Alfred Q&A] [answered]
When I looked up the [docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.input.TextArea.html) about TextArea inputs, it said:

<html>
<body>

align | Horizontal alignment of text in the element. Does not usually apply to labels. Think of it as an alignment that applies to the innermost layer of an element (while labels are generally located at outer layers). See LabelledElement for more on labelled elements. Can be left (default), center, right, or justify.
-- | --

</body>
</html>

So the align attribute does not change the align of the label. So how do I change that instead?

**Comment by storyteller-dev** [comment]
Instead of changing the inputs align attribute, you get its label and change the align of the label directly:


```python
import alfred3 as al

exp = al.Experiment()


@exp.member
class DemoPage(al.Page):
    def on_exp_access(self):
        text_area = al.TextArea(
            toplab="Textarea Label",
            name="my_textarea",
            align="left", # Does not change the label align!
            force_input=True,
        )
        text_area.toplab.align = "left"
        self += text_area
```

**Reply by jobrachem** [reply]
This is a nice solution that cleverly uses alfred3's object-oriented capabilities. You can also achieve the same result without having to first create the `text_area` object and then changing its `toplab` attribute by directly using a `alfred3.Label` in the definition of the toplab (see second answer).

**Comment by jobrachem** [comment, marked as answer]
Instead of using a bare string as element label, you can use the `alfred3.Label` class ([docs](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.element.display.Label.html?highlight=label#alfred3.element.display.Label)) to get full control over your labels.

Here's a minimal example:

```python
import alfred3 as al

exp = al.Experiment()


@exp.member
class DemoPage(al.Page):

    def on_exp_access(self):
        self += al.TextArea(
            toplab=al.Label("Textarea Label", align="left"),
            name="my_textarea",
            align="left", # Does not change the label align!
            force_input=True,
        )
```
